### PR TITLE
admission: enable webhooks by default

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of https://github.com/openshift/origin/pull/20743.

### CODE REVIEW

**Comment by deads2k** [comment]
/lgtm

/hold 

hold for prereq pull.

File: pkg/cmd/server/origin/admission/register.go [hunk 1]
```diff
@@ -25,7 +20,13 @@ import (
 	quotarunonceduration "github.com/openshift/origin/pkg/quota/apiserver/admission/runonceduration"
 	schedulerpodnodeconstraints "github.com/openshift/origin/pkg/scheduler/admission/podnodeconstraints"
 	securityadmission "github.com/openshift/origin/pkg/security/apiserver/admission/sccadmission"
+	mutatingwebhook "k8s.io/apiserver/pkg/admission/plugin/webhook/mutating"
+	validatingwebhook "k8s.io/apiserver/pkg/admission/plugin/webhook/validating"
 
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/admission"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	kubeapiserver "k8s.io/kubernetes/pkg/kubeapiserver/options"
```
Comment (simo5) [inline pkg/cmd/server/origin/admission/register.go]: Shouldn't vendored dependencies be called before openshift ones ?
At least that's what I see everywhere else

File: pkg/project/apiserver/admission/nodeenv/admission.go [hunk 2]
```diff
@@ -31,9 +31,11 @@ type podNodeEnvironment struct {
 
 var _ = oadmission.WantsProjectCache(&podNodeEnvironment{})
 var _ = kadmission.WantsInternalKubeClientSet(&podNodeEnvironment{})
+var _ = admission.ValidationInterface(&podNodeEnvironment{})
+var _ = admission.MutationInterface(&podNodeEnvironment{})
 
 // Admit enforces that pod and its project node label selectors matches at least a node in the cluster.
-func (p *podNodeEnvironment) Admit(a admission.Attributes) (err error) {
+func (p *podNodeEnvironment) admit(a admission.Attributes, mutationAllowed bool) (err error) {
```
Comment (simo5) [inline pkg/project/apiserver/admission/nodeenv/admission.go]: The point of the boolean is to change behavior from mutation to validation only.
If you call the variable "validationOnly" then the code flow below can be changed from:
```
if mutationAllowed {
    ...
} else if (...) {
    return xyz
}
...
```
to a more readable:
```
if validationOnly {
    return xyz
}
... /* mutation and all */
```

At least I find this more readable and natural.

File: pkg/cmd/server/origin/admission/register.go (hunk 1 above)
Comment (sttts) [reply pkg/cmd/server/origin/admission/register.go]: fixed

File: pkg/project/apiserver/admission/nodeenv/admission.go (hunk 2 above)
Comment (sttts) [reply pkg/project/apiserver/admission/nodeenv/admission.go]: We have this mutationAllowed style at a number of places. Would avoid to change that.

But I updated the code to follow your if-clause order, makes it easier, compare https://github.com/openshift/origin/pull/20743.

**Comment by sttts** [comment]
Pre-req is green. Rebasing.

**Comment by sttts** [comment]
Prereq is in.

/hold cancel

**Pull request merged**